sql/information_schema/information_schema.go

```
@@ -1967,6 +1972,164 @@ func (pit *informationSchemaPartitionIter) Close(_ *Context) error {
+	s, ok := n.stats[DbTable{Db: db, Table: table}]
+	if ok {
+		return s.RowCount, nil
+	}
+
+	t, _, err := n.catalog.Table(ctx, db, table)
+	if err != nil {
+		return 0, err
+	}
+	st, ok := t.(StatisticsTable)
+	if !ok {
+		return 0, nil
+	}
+	return st.RowCount(ctx)
+}
+
+func (n *defaultStatsTable) Analyze(ctx *Context, db, table string) error {
+	tableStats := &TableStatistics{
+		CreatedAt: time.Now(),
+	}
+
+	t, _, err := n.catalog.Table(ctx, db, table)
+	if err != nil {
+		return err
+	}
+	histMap, err := NewHistogramMapFromTable(ctx, t)
+	if err != nil {
+		return err
+	}
+
+	tableStats.Histograms = histMap
+	for _, v := range histMap {
+		tableStats.RowCount = v.Count + v.NullCount
+		break
+	}
+
+	n.stats[DbTable{Db: db, Table: table}] = tableStats
+	return nil
+}
+
+func newUpdatableStatsTable() *updatableStatsTable {
+	return &updatableStatsTable{
+		defaultStatsTable: NewDefaultStats(),
+	}
+}
+
+// updatableStatsTable provides a statistics table that can
+// be edited with UPDATE statements.
+type updatableStatsTable struct {
+	*defaultStatsTable
+}
+
+var _ UpdatableTable = (*updatableStatsTable)(nil)
+var _ StatsReadWriter = (*updatableStatsTable)(nil)
+
+// AssignCatalog implements sql.CatalogTable
+func (t *updatableStatsTable) AssignCatalog(cat Catalog) Table {
+	t.catalog = cat
+	return t
+}
+
+// Updater implements sql.UpdatableTable
+func (t *updatableStatsTable) Updater(_ *Context) RowUpdater {
+	return newStatsEditor(t.catalog, t.stats)
+}
+
+func newStatsEditor(c Catalog, stats map[DbTable]*TableStatistics) RowUpdater {
+	return &statsEditor{c: c, s: stats}
+}
+
+type statsEditor struct {
+	c Catalog
+	s map[DbTable]*TableStatistics
+}
+
+var _ RowUpdater = (*statsEditor)(nil)
+
+// StatementBegin implements sql.RowUpdater
+func (s *statsEditor) StatementBegin(ctx *Context) {}
+
+// DiscardChanges implements sql.RowUpdater
+func (s *statsEditor) DiscardChanges(ctx *Context, errorEncountered error) error { return nil }
```

Make this always throw an error instead, will lead so silent errors otherwise